### PR TITLE
Implemented BMInterface functions required for rejecting games

### PR DESCRIPTION
**Pull request by blackshadowshade**
Addresses #205.

This change should be invisible to players.

http://jenkins.buttonweavers.com:8080/job/buttonmen-blackshadowshade/591/

### CODE REVIEW

File: src/engine/BMInterface.php [hunk 1]
```diff
@@ -605,6 +613,38 @@ protected function retrieve_button_ids($playerIdArray, $buttonNameArray) {
         return $buttonIdArray;
     }
 
+    public function save_join_game_decision($playerId, $gameId, $decision) {
+        if (!is_bool($decision)) {
+            throw new InvalidArgumentException('decision must be boolean');
+        }
```
Comment (cgolubi1) [inline src/engine/BMInterface.php]: Editorial comment: for the most part, the other public BMInterface() functions take text arguments like 'add' or 'decline'.  It seems weird that this one should take a boolean, and, given the little API inconsistencies we already have with arguments which are passed as booleans (there are a couple of outstanding technical debt items about this), i'm lukewarm on booleans as being a simple interface.  I think we should expect that the API function for whether or not to join a game will pass a string, and that apiResponder.php will pass that string on to BMInterface as a string.

**Comment by cgolubi1** [comment]
Other than that, in terms of testing this:
* It looks like all existing games on dev and prod have `has_player_accepted=1` for both players.  However, this is definitely a pull for which i'll want to populate a system with a number of games in different states of play (open, open/random button, in-progress in a few places, finished), then update it, and basically visually look for regressions in how those games look.
* I might as well throw some random games and random/regression games at the system just in case, but realistically RandomAI only tests one path between start and a playable game, so i think it's unlikely that i'll catch much with those, and am not going to test heavily that way.  If i find anything, it's going to be by trying creating games a bunch of different ways and looking at them.

**Comment by blackshadowshade** [comment]
So, I've changed the input parameter from a boolean to a string, as requested.

http://jenkins.buttonweavers.com:8080/job/buttonmen-blackshadowshade/607/

**Comment by blackshadowshade** [comment]
Sorry, added the change to the QUnit test for copyright year so that I can get a clean Jenkins run.

http://jenkins.buttonweavers.com:8080/job/buttonmen-blackshadowshade/609/

**Comment by cgolubi1** [comment]
I don't see any problems in manual testing.  Merging.

**Pull request merged**